proxy/src/transport/tls/dns_name.rs

```diff
@@ -0,0 +1,30 @@
+use super::webpki;
+use std::fmt;
+use convert::TryFrom;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct DnsName(webpki::DNSName);
```

Comment:
Could you please include a comment on a type with a brief description. Ideally, I'd include a link to the relevant RFC that describes the validation that's implemented.

Reply:
> Could you please include a comment on a type with a brief description. Ideally, I'd include a link to the relevant RFC that describes the validation that's implemented.

Added!